### PR TITLE
fix: don't block the connection during the scram dance

### PR DESCRIPTION
**Pull request by abc3**

### CODE REVIEW

**Comment by chasers** [comment]
Hmmm failing test?

**Comment by abc3** [comment]
Yes, I've found a race condition and am fixing it right now

File: lib/supavisor/client_handler.ex [hunk 1]
```diff
@@ -383,24 +384,24 @@ defmodule Supavisor.ClientHandler do
 
     case Db.call(db_pid, self(), bin) do
       :ok ->
-        Logger.debug("DB call success")
+        Logger.debug("db_handler call success")
```
Comment (chasers) [inline lib/supavisor/client_handler.ex:387]:
```suggestion
        Logger.debug("DbHandler call success")
```

File: lib/supavisor/client_handler.ex [hunk 2]
```diff
@@ -383,24 +384,24 @@ defmodule Supavisor.ClientHandler do
...
         :keep_state_and_data
 
       {:buffering, size} ->
-        Logger.debug("DB call buffering #{size}")
+        Logger.debug("db_handler call buffering #{size}")
```
Comment (chasers) [inline lib/supavisor/client_handler.ex:391]:
```suggestion
        Logger.debug("DbHandler call buffering #{size}")
```

File: lib/supavisor/client_handler.ex [hunk 3]
```diff
@@ -383,24 +384,24 @@ defmodule Supavisor.ClientHandler do
...
 
         if size > 1_000_000 do
-          msg = "Db buffer size is too big: #{size}"
+          msg = "db_handler buffer size is too big: #{size}"
```
Comment (chasers) [inline lib/supavisor/client_handler.ex:394]:
```suggestion
          msg = "DbHandler buffer size is too big: #{size}"
```

File: lib/supavisor/client_handler.ex [hunk 4]
```diff
@@ -383,24 +384,24 @@ defmodule Supavisor.ClientHandler do
...
           Logger.error(msg)
           HH.sock_send(data.sock, Server.error_message("XX000", msg))
           {:stop, :normal}
         else
-          Logger.debug("DB call buffering")
+          Logger.debug("db_handler call buffering")
```
Comment (chasers) [inline lib/supavisor/client_handler.ex:399]:
```suggestion
          Logger.debug("DbHandler call buffering")
```

File: lib/supavisor/client_handler.ex [hunk 5]
```diff
@@ -383,24 +384,24 @@ defmodule Supavisor.ClientHandler do
...
           :keep_state_and_data
         end
 
       {:error, reason} ->
-        msg = "DB call error: #{inspect(reason)}"
+        msg = "db_handler error: #{inspect(reason)}"
```
Comment (chasers) [inline lib/supavisor/client_handler.ex:404]:
```suggestion
        msg = "DbHandler error: #{inspect(reason)}"
```

File: lib/supavisor/db_handler.ex [hunk 6]
```diff
@@ -14,15 +14,20 @@ defmodule Supavisor.DbHandler do
...
     :gen_statem.start_link(__MODULE__, config, hibernate_after: 5_000)
   end
 
   @spec call(pid(), pid(), binary()) :: :ok | {:error, any()} | {:buffering, non_neg_integer()}
-  def call(pid, caller, msg) do
-    :gen_statem.call(pid, {:db_call, caller, msg}, 15_000)
-  end
+  def call(pid, caller, msg), do: :gen_statem.call(pid, {:db_call, caller, msg}, 15_000)
+
+  @spec get_state(pid()) :: any()
```
Comment (chasers) [inline lib/supavisor/db_handler.ex:26]: Should this be a list of possibe states as atoms?

Comment (abc3) [reply lib/supavisor/db_handler.ex:26]: yeah, that would be more readable

**Review by chasers — APPROVED** [review]

**Comment by chasers** [comment]
Couple small things but otherwise looks fine.

**Pull request merged**